sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/SecretsRegexTest.java

```diff
@@ -0,0 +1,284 @@
+import org.sonar.java.checks.regex.PossessiveQuantifierContinuationCheck;
+import org.sonar.java.checks.regex.RedosCheck;
+import org.sonar.java.checks.regex.RedundantRegexAlternativesCheck;
+import org.sonar.java.checks.regex.RegexComplexityCheck;
+import org.sonar.java.checks.regex.RegexLookaheadCheck;
+import org.sonar.java.checks.regex.RegexStackOverflowCheck;
+import org.sonar.java.checks.regex.ReluctantQuantifierCheck;
+import org.sonar.java.checks.regex.ReluctantQuantifierWithEmptyContinuationCheck;
+import org.sonar.java.checks.regex.SingleCharCharacterClassCheck;
+import org.sonar.java.checks.regex.SingleCharacterAlternationCheck;
+import org.sonar.java.checks.regex.SuperfluousCurlyBraceCheck;
+import org.sonar.java.checks.regex.UnicodeAwareCharClassesCheck;
+import org.sonar.java.checks.regex.UnicodeCaseCheck;
+import org.sonar.java.checks.regex.UnquantifiedNonCapturingGroupCheck;
+import org.sonar.java.checks.regex.UnusedGroupNamesCheck;
+import org.sonar.java.checks.regex.VerboseRegexCheck;
+import org.sonar.java.model.InternalSyntaxToken;
+import org.sonar.java.model.expression.IdentifierTreeImpl;
+import org.sonar.java.model.expression.LiteralTreeImpl;
+import org.sonar.java.model.expression.MemberSelectExpressionTreeImpl;
+import org.sonar.java.model.expression.MethodInvocationTreeImpl;
+import org.sonar.java.regex.JavaAnalyzerRegexSource;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.secrets.api.SpecificationLoader;
+import org.sonar.plugins.secrets.configuration.model.Rule;
+import org.sonar.plugins.secrets.configuration.model.matching.AuxiliaryPattern;
+import org.sonar.plugins.secrets.configuration.model.matching.BooleanCombination;
+import org.sonar.plugins.secrets.configuration.model.matching.Match;
+import org.sonarsource.analyzer.commons.regex.RegexParser;
+import org.sonarsource.analyzer.commons.regex.ast.FlagSet;
+
+class SecretsRegexTest {
+
+  private final static String CONFIGURATION_FILES_PATH = String.join(File.separator, "src", "main", "resources", "org", "sonar", "plugins", "secrets", "configuration");
+
+  private final TestRegexScannerContext context = new TestRegexScannerContext();
+
+  @Test
+  @Disabled("The issues should be review")
+  void shouldValidateAllRegexesInConfiguration() {
+    Set<String> listOfFileNames = listOfYamlFiles();
+
+    List<PatternLocation> patternLocations = convertToPatternLocations(listOfFileNames);
+
+    patternLocations.forEach(patternLocation -> checkRegex(patternLocation, patternLocation.secretRspecKey));
+
+    context.verify();
+  }
+
+  @Test
+  @Disabled("For simple run on single YAML file")
+  void shouldValidateSingleFile() {
+    List<PatternLocation> patternLocations = convertToPatternLocations("google-oauth2.yaml");
```

Comment:
Maybe this should be made configurable?
If I could run `mvn -Dtest=SecretsRegexTest#shouldValidateSingleFile -DargLine="-Dfilename=google-oauth2.yaml"`, this would be nice.

Reply:
Is it not easier to change line 102 and run the test directly from IDE? 🤔

Reply:
For us yes, for AppSec I believe no.